.circleci/config.yml

```diff
@@ -9,36 +9,65 @@ checkout:
         git checkout -qf FETCH_HEAD
       fi
 
-jobs:
 
-  unittests:
+executors:
+  py27:
+    docker:
+      - image: circleci/python:2.7
+
+  py36:
+    docker:
+      - image: circleci/python:3.6
+
+  centos7:
     docker:
-      - image: circleci/python:2.7.15-stretch
+      - image: amd64/centos:centos7.3.1611
+
+  cloudify-machine:
+    machine:
+      image: ubuntu-1604:201903-01
+
+
+commands:
+  run_unittest:
     steps:
       - checkout
       - run:
-          name: Download pip
-          command: curl "https://bootstrap.pypa.io/get-pip.py" -o "get-pip.py"
+          name: install tox
+          command: pip install --user tox
       - run:
-          name: Install pip
-          command: sudo python get-pip.py
+          name: Run Flake8
+          command: /home/circleci/.local/bin/tox -e pep8
       - run:
-          name: Install virtualenv
-          command: pip install --user virtualenv==15.1.0
+         name: Run nosetest
+         command: /home/circleci/.local/bin/tox -e nosetest
+
+  check_py3_compat:
+    steps:
       - run:
-          name: Init virtualenv
-          command: virtualenv env && source env/bin/activate
+          name: install futurize
+          command: pip install future --user
       - run:
-          name: install tox
-          command: pip install --user tox
-      - run: /home/circleci/.local/bin/tox -e pep8
-      - run: /home/circleci/.local/bin/tox -e py27
+          name: find python3-incompatible code
+          command: |
+            FUTURIZE="futurize ."
+            while read line; do
+              [[ "$line" =~ ^#.* ]] && continue
+              FUTURIZE="${FUTURIZE} ${line}"
+            done<.circleci/py3fixers
+            echo "Running: $FUTURIZE"
+            $FUTURIZE>futurize_diffs
+      - run:
+          name: check that there is no python3-incompatible code
+          command: |
+            if [[ -s futurize_diffs ]]; then
+              echo "Python-3-incompatible code found"
+              cat futurize_diffs
+              exit 1
+            fi
 
-  wagon:
-    docker:
-      - image: amd64/centos:centos7.3.1611
+  generate_wagon:
```

Comment:
Well, the reason why I named them this way because this refer to verb/actions which are the commands, if you check the the executor section you will notice that I call the user defined executor as wagon.

So `generate_wagon` refers to the list of steps(commands) that should be executed on executor `wagon`

let me know what do you think ?